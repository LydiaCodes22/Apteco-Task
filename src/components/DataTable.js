/* eslint-disable react/prop-types */
import React from "react";

const DataTable = ({ columnTitles, rowTitles, data }) => {
  return (
    <>
      <h2>table</h2>
      <table>
        <tbody>
          <tr>
            {React.Children.toArray(
              columnTitles.map((item) => {
                return <th>{item}</th>;
              })
            )}
          </tr>
          {React.Children.toArray(
            rowTitles.map((item) => {
              return (
                <tr>
                  <th>{item}</th>
                  {React.Children.toArray(
                    data[rowTitles.indexOf(item)].map((x) => {
                      return <td>{x}</td>;
                    })
                  )}
                </tr>
              );
            })
          )}
          ;
        </tbody>
      </table>
    </>
  );
};

export default DataTable;
