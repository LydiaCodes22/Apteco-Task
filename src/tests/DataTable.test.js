import { render, screen } from "@testing-library/react";
import React from "react";
import DataTable from "../components/DataTable";

describe("the data table", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<DataTable />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the correct place names", () => {
    render(<DataTable />);
    expect(screen.getByText(/australia/i)).toBeInTheDocument();
    expect(screen.getByText(/New Zealand/i)).toBeInTheDocument();
    expect(screen.getByText(/Denmark/i)).toBeInTheDocument();
    expect(screen.getByText(/France/i)).toBeInTheDocument();
    expect(screen.getByText(/Germany/i)).toBeInTheDocument();
    expect(screen.getByText(/Greece/i)).toBeInTheDocument();
    expect(screen.getByText(/italy/i)).toBeInTheDocument();
    expect(screen.getByText(/Latvia/i)).toBeInTheDocument();
    expect(screen.getByText(/Portugal/i)).toBeInTheDocument();
    expect(screen.getByText(/Sweden/i)).toBeInTheDocument();
    expect(screen.getByText(/Jamaica/i)).toBeInTheDocument();
    expect(screen.getByText(/United States/i)).toBeInTheDocument();
    expect(screen.getByText(/Kuwait/i)).toBeInTheDocument();
    expect(screen.getByText(/Mongolia/i)).toBeInTheDocument();
    expect(screen.getByText(/Mali/i)).toBeInTheDocument();
    expect(screen.getByText(/Namibia/i)).toBeInTheDocument();
    expect(screen.getByText(/Senegal/i)).toBeInTheDocument();
    expect(screen.getByText(/sierra Leone/i)).toBeInTheDocument();
    expect(screen.getByText(/South Africa/i)).toBeInTheDocument();
  });
  it("renders the correct years", () => {
    render(<DataTable />);
    expect(screen.getByText(/2018/i)).toBeInTheDocument();
    expect(screen.getByText(/2019/i)).toBeInTheDocument();
    expect(screen.getByText(/2020/i)).toBeInTheDocument();
    expect(screen.getByText(/2021/i)).toBeInTheDocument();
    expect(screen.getByText(/2022/i)).toBeInTheDocument();
    expect(screen.getByText(/2023/i)).toBeInTheDocument();
  });
});
