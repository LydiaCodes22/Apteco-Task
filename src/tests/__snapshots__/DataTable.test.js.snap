// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the data table matches the snapshot 1`] = `
<DocumentFragment>
  <h2>
    table
  </h2>
  <table>
    <tbody>
      <tr>
        <th>
          Unclassified
        </th>
        <th>
          Australia
        </th>
        <th>
          New Zealand
        </th>
        <th>
          Denmark
        </th>
        <th>
          France
        </th>
        <th>
          Germany
        </th>
        <th>
          Greece
        </th>
        <th>
          Italy
        </th>
        <th>
          Latvia
        </th>
        <th>
          Portugal
        </th>
        <th>
          Sweden
        </th>
        <th>
          Jamaica
        </th>
        <th>
          United States
        </th>
        <th>
          Kuwait
        </th>
        <th>
          Mongolia
        </th>
        <th>
          Mali
        </th>
        <th>
          Namibia
        </th>
        <th>
          Senegal
        </th>
        <th>
          Sierra Leone
        </th>
        <th>
          South Africa
        </th>
        <th>
          iTOTAL
        </th>
      </tr>
      <tr>
        <th>
          Unclassified
        </th>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <th>
          2018
        </th>
        <td>
          6697
        </td>
        <td>
          28001
        </td>
        <td>
          93
        </td>
        <td>
          158
        </td>
        <td>
          51822
        </td>
        <td>
          24702
        </td>
        <td>
          10995
        </td>
        <td>
          2665
        </td>
        <td>
          62
        </td>
        <td>
          2187
        </td>
        <td>
          2457
        </td>
        <td>
          19
        </td>
        <td>
          30436
        </td>
        <td>
          2356
        </td>
        <td>
          24
        </td>
        <td>
          103
        </td>
        <td>
          241
        </td>
        <td>
          85
        </td>
        <td>
          465
        </td>
        <td>
          87
        </td>
        <td>
          125238
        </td>
      </tr>
      <tr>
        <th>
          2019
        </th>
        <td>
          7904
        </td>
        <td>
          32043
        </td>
        <td>
          108
        </td>
        <td>
          168
        </td>
        <td>
          60101
        </td>
        <td>
          30059
        </td>
        <td>
          12771
        </td>
        <td>
          3191
        </td>
        <td>
          67
        </td>
        <td>
          2471
        </td>
        <td>
          2911
        </td>
        <td>
          14
        </td>
        <td>
          36561
        </td>
        <td>
          2745
        </td>
        <td>
          17
        </td>
        <td>
          122
        </td>
        <td>
          250
        </td>
        <td>
          99
        </td>
        <td>
          612
        </td>
        <td>
          119
        </td>
        <td>
          151455
        </td>
      </tr>
      <tr>
        <th>
          2020
        </th>
        <td>
          16108
        </td>
        <td>
          66702
        </td>
        <td>
          224
        </td>
        <td>
          352
        </td>
        <td>
          125551
        </td>
        <td>
          60345
        </td>
        <td>
          26639
        </td>
        <td>
          6553
        </td>
        <td>
          147
        </td>
        <td>
          5049
        </td>
        <td>
          6116
        </td>
        <td>
          41
        </td>
        <td>
          75735
        </td>
        <td>
          5512
        </td>
        <td>
          38
        </td>
        <td>
          214
        </td>
        <td>
          536
        </td>
        <td>
          184
        </td>
        <td>
          1078
        </td>
        <td>
          207
        </td>
        <td>
          311775
        </td>
      </tr>
      <tr>
        <th>
          2021
        </th>
        <td>
          11045
        </td>
        <td>
          45946
        </td>
        <td>
          113
        </td>
        <td>
          242
        </td>
        <td>
          85783
        </td>
        <td>
          54356
        </td>
        <td>
          18346
        </td>
        <td>
          5023
        </td>
        <td>
          97
        </td>
        <td>
          3531
        </td>
        <td>
          4110
        </td>
        <td>
          32
        </td>
        <td>
          62853
        </td>
        <td>
          3893
        </td>
        <td>
          15
        </td>
        <td>
          135
        </td>
        <td>
          381
        </td>
        <td>
          167
        </td>
        <td>
          787
        </td>
        <td>
          164
        </td>
        <td>
          255218
        </td>
      </tr>
      <tr>
        <th>
          2022
        </th>
        <td>
          12946
        </td>
        <td>
          53780
        </td>
        <td>
          178
        </td>
        <td>
          298
        </td>
        <td>
          102409
        </td>
        <td>
          63754
        </td>
        <td>
          21540
        </td>
        <td>
          5937
        </td>
        <td>
          89
        </td>
        <td>
          4143
        </td>
        <td>
          4927
        </td>
        <td>
          38
        </td>
        <td>
          72556
        </td>
        <td>
          4534
        </td>
        <td>
          32
        </td>
        <td>
          209
        </td>
        <td>
          465
        </td>
        <td>
          146
        </td>
        <td>
          936
        </td>
        <td>
          199
        </td>
        <td>
          297910
        </td>
      </tr>
      <tr>
        <th>
          2023
        </th>
        <td>
          12312
        </td>
        <td>
          51162
        </td>
        <td>
          158
        </td>
        <td>
          284
        </td>
        <td>
          97846
        </td>
        <td>
          62683
        </td>
        <td>
          20155
        </td>
        <td>
          5861
        </td>
        <td>
          112
        </td>
        <td>
          3995
        </td>
        <td>
          4670
        </td>
        <td>
          34
        </td>
        <td>
          71055
        </td>
        <td>
          4396
        </td>
        <td>
          28
        </td>
        <td>
          187
        </td>
        <td>
          426
        </td>
        <td>
          158
        </td>
        <td>
          902
        </td>
        <td>
          178
        </td>
        <td>
          290359
        </td>
      </tr>
      <tr>
        <th>
          iTOTAL
        </th>
        <td>
          67012
        </td>
        <td>
          274857
        </td>
        <td>
          870
        </td>
        <td>
          1497
        </td>
        <td>
          522932
        </td>
        <td>
          214163
        </td>
        <td>
          109564
        </td>
        <td>
          25958
        </td>
        <td>
          551
        </td>
        <td>
          21230
        </td>
        <td>
          25175
        </td>
        <td>
          178
        </td>
        <td>
          273879
        </td>
        <td>
          23388
        </td>
        <td>
          154
        </td>
        <td>
          968
        </td>
        <td>
          2276
        </td>
        <td>
          777
        </td>
        <td>
          4778
        </td>
        <td>
          954
        </td>
        <td>
          1156553
        </td>
      </tr>
    </tbody>
  </table>
  ;
</DocumentFragment>
`;
